01_fib/cuminandpaprika/main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(7)
+}
+
+// Print n number of fibonnaci sequence
+// 1, 1, 2, 3, 5, 8, 13
+// Note: we take the absolute value of n to ensure our loop works.
+func fib(n int16) {
+	var i int16
+	if isPositive(n) {
+		for ; i < n; i++ {
+			fmt.Fprintln(out, calcFib(i))
+		}
+		return
+	}
+
+	for ; i > n; i-- {
+		fmt.Fprintln(out, calcFib(i))
+	}
+
+}
+
+func calcFib(n int16) int16 {
+	if isPositive(n) {
```

Comment:
nitpick again: Love that you broke this down to nested function calls, but `calcFib()` seems redundant given you've unwrapped the loop above in `fib()`, basically doing the same check twice?

Reply:
That's completely right, must've been left over from when I refactored. Removed this redundant function, good pick up